src/NuGet.Core/NuGet.ProjectManagement/Projects/BuildIntegratedNuGetProject.cs

```diff
@@ -54,6 +49,22 @@ public abstract class BuildIntegratedNuGetProject : NuGetProject, INuGetIntegrat
             INuGetProjectContext projectContext,
             bool throwOnFailure);
 
+        public abstract Task<bool> InstallPackageAsync(
+            PackageIdentity packageIdentity,
+            INuGetProjectContext nuGetProjectContext,
+            IEnumerable<NuGetFramework> successfulFrameworks,
+            IEnumerable<NuGetFramework> unsucessfulFrameworks,
```

Comment:
`NuGetPackageManager` will always have something from the project action to pass in here. An alternative would be to not have this overload on Legacy and P.J. (only on P.R.). I'll try that.